src/main/kotlin/org/rust/ide/navigation/goto/RustGotoSuperHandler.kt

```diff
@@ -4,20 +4,36 @@ import com.intellij.lang.LanguageCodeInsightActionHandler
 import com.intellij.openapi.editor.Editor
 import com.intellij.openapi.project.Project
 import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RustImplItemElement
+import org.rust.lang.core.psi.RustImplMethodMemberElement
 import org.rust.lang.core.psi.RustMod
+import org.rust.lang.core.psi.RustNamedElement
 import org.rust.lang.core.psi.impl.RustFile
 import org.rust.lang.core.psi.util.parentOfType
+import org.rust.lang.core.psi.util.trait
 
 class RustGotoSuperHandler : LanguageCodeInsightActionHandler {
     override fun startInWriteAction() = false
 
     override fun invoke(project: Project, editor: Editor, file: PsiFile) {
         val focusedElement = file.findElementAt(editor.caretModel.offset) ?: file ?: return
 
-        val focusedMod = focusedElement.parentOfType<RustMod>(strict = false) ?: return
-        val superMod = focusedMod.`super` ?: return
+        val focusedMethod = focusedElement.parentOfType<RustImplMethodMemberElement>(strict = false)
+        if (focusedMethod != null) {
+            (getSuperMethod(focusedMethod) ?: return).navigate(true)
+        } else {
+            val focusedMod = focusedElement.parentOfType<RustMod>(strict = false) ?: return
+            val superMod = focusedMod.`super` ?: return
 
-        superMod.navigate(true)
+            superMod.navigate(true)
+        }
+    }
+
+    private fun getSuperMethod(focusedMethod: RustImplMethodMemberElement): RustNamedElement? {
+        val rustImplItem = focusedMethod.parentOfType<RustImplItemElement>(strict = false) ?: return null
```

Comment:
No need in `strict = false` here, because method and impl can never be the same PSI element. 

Also perhaps it makes sense to introduce an extension methods to get particular ancestor elements in a typed manner? Something like `methodImpl.imp`, `methodDefinition.trait`. Currently we use ad-hoc `parentOfType` and chains of `.parent` everywhere, looks like this is not the best approach.